Changes for jira PUBSTWO-1689 (Extend people lookups):

orosz-usgs opened the pull request:
Updated /lookup/people end point to show View.PW json fields
   Refactored endpoint to use PersonContributorFilterParam for query parameters
   Updated tests with new contributor search parameters
   Added new exception: PubsInvalidParameterException

Commit:
Changes for jira PUBSTWO-1689 (Extend people lookups):
   Updated /lookup/people end point to show View.PW json fields
   Refactored endpoint to use PersonContributorFilterParam for query parameters
   Updated tests with new contributor search parameters
   Added new exception: PubsInvalidParameterException

Commit:
Changes for jira PUBSTWO-1689 (Extend people lookups):
   Removed validation from /lookup/people endpoint, the need to include validation will
be addressed in a later ticket.
   Removed setCorporation and setUsgs from Contributor class
   Moved definition of contant CORPORATION to CorporateContributorDao
   Updated tests

dsteinich commented on src/main/resources/mybatis/mappers/contributor.xml:
```diff
@@ -101,18 +101,27 @@
         </if>
-        <if test="given != null">
-            and upper(given) like upper(#{given}) || '%'
+        <if test="given != null and given.length > 0">
```
I think the changes in this file will break logic in other places (anywhere else that getByMap, getByMapPersonContributor, or getByMapCorporateContributor is used for any flavor of contributor). It would appear that I did not create IT tests needed to catch the regression.

dsteinich replied:
ExtPublicationContributorService's 
protected OutsideContributor getByName(String family, String given)
method will not work with this change

dsteinich commented on src/test/java/gov/usgs/cida/pubs/webservice/LkupMvcServiceTest.java:
```diff
@@ -52,6 +53,7 @@ public void setup() {
 		personContributor = new UsgsContributor();
 	}
 
+	@Ignore // TODO: refactor test for pojo query parameters
```
Please include this refactor now.

Commit:
Changes for jira PUBSTWO-1689 (Extend people lookups):
Test Cleanup:
  Implemented Ignored test
  Add contributor to test data set.
  Added test for getting by contributor.xml filters to CorporateContributorDaoIT
  Cleaned up PersonContributorDaoIT tests that depended on the id order in returned list

dsteinich commented on src/test/java/gov/usgs/cida/pubs/dao/PersonContributorDaoIT.java:
```diff
@@ -187,42 +188,32 @@ public void getObjectCount() {
-		assertEquals(44, contributors.get(10).getId().intValue());
-		assertEquals(3, contributors.get(11).getId().intValue());
-		assertEquals(2, contributors.get(12).getId().intValue());
+		List<Integer> ids = ContributorDaoIT.getIds(contributors);
```
These tests are to show that the 'order by' in the query is working.

dsteinich commented on src/test/java/gov/usgs/cida/pubs/dao/PersonContributorDaoIT.java:
```diff
@@ -187,42 +188,32 @@ public void getObjectCount() {
-		assertEquals(24, contributors.get(2).getId().intValue());
-		assertEquals(34, contributors.get(3).getId().intValue());
-		assertEquals(44, contributors.get(4).getId().intValue());
+		List<Integer> ids = ContributorDaoIT.getIds(contributors);
```
These tests are to show that the 'order by' in the query is working.

dsteinich commented on src/test/java/gov/usgs/cida/pubs/dao/PersonContributorDaoIT.java:
```diff
@@ -187,42 +188,32 @@ public void getObjectCount() {
-		assertEquals(2, contributors.size());
-		assertEquals(1, contributors.get(0).getId().intValue());
-		assertEquals(11, contributors.get(1).getId().intValue());
+		List<Integer> ids = ContributorDaoIT.getIds(contributors);
```
These tests are to show that the 'order by' in the query is working.

Commit:
Changes for jira PUBSTWO-1689 (Extend people lookups):
  Restored tests in PersonContributorDaoIT to check for the order returned in query.
  Updated test getByPreferredAll to check for the new test contributor.

Commit:
Changes for jira PUBSTWO-1689 (Extend people lookups):
  Added new IT test : ExtPublicationContributorServiceIT
  Passed running against master.
  This showed bug in ExtPublicationContributorServiceIT.getByName() in branch.
  Fixed error in ExtPublicationContributorServiceIT.getByName()
